src/hints.rs

```diff
@@ -286,6 +287,20 @@ impl From<&NotificationHint> for (MessageItem, MessageItem) {
     }
 }
 
+#[deprecated(note = "To convert a key-value pair of MessageItems, use From<(&MessageItem, &MessageItem)>")]
+#[cfg(all(unix, not(target_os = "macos")))]
+impl From<&MessageItem> for NotificationHint {
```

Comment:
I hope this was not actually externally used by anybody anyway, you don't have to bend over backwards to make it compatible though. I'm really gonna make a 4.0.0 release anyway I will probably also hide a few implementation details if I can